### DISCUSSION TITLE
Probleme mit Installation/Update des EBUS5-Adapters

### DISCUSSION THREAD

**Discussion by cebulonde** [john30/ebusd#1136] [Q&A]
Hallo,

ich habe die Absicht, eine Vaillant ecoCOMPAKT-Heizung mit einer calorMATIC 430 an ein HomeMatic-IP-System (RaspberryMatic) anzubinden, ggf. unter Verwendung des ioBroker. Also habe ich den eBUS Adapter 5 gekauft, Version WiFi, Auslieferung 10.2023. Leider bin ich bezüglich dieser Kopplung ein absoluter Anfänger und scheitere bereits an sehr grundsätzlichen Dingen.

Meine Randbedingungen: Windows 10 PC, aktuelle SW-Version, Browser Microsoft Edge, Adapter noch NICHT am EBUS der Heizung angeschlossen.

Ich habe ausgelieferte Version des EBUS5-Adapters mit Hilfe von Web Serial auf die Version 20131223 upgedated. Danach via WiFi Access Point die Parameter auf mein lokales WLAN eingestellt. Trotzdem keine Verbindung zum Router erreicht. Der Adapter taucht als serielles USB-Gerät (COM4) im Gerätemanager des PC auf, aber NICHT als WLAN-Gerät im Router.

Vor einigen Tagen die Firmware-Version 20240106 entdeckt. Jeglicher Versuch, die neue Version zu installieren, scheitert, auch wenn ich den Boot-Button beim Anklicken des Install-Kommandos gedrückt halte.

Die Meldung lautet: `Failed to initialize. Try resetting your device or holding the BOOT button while clicking INSTALL. 
`
Das gilt auch bei der vergeblichen Anwendung des Erase-Tools.

Ein Ansprechen des Adapters über den WiFi Access Point funktioniert immer noch: 

```
Build: [20231223](https://adapter.ebusd.eu/v5/ChangeLog#20231223)
[ebusd device string](https://github.com/john30/ebusd/wiki/2.-Run#device-options): ens:/dev/ttyACM0 (number may vary)
WiFi station/client: inactive
WiFi access point: EBUS, 192.168.4.1
Chip ID: 58cf79e48a70, [ESP32-C3, rev 3](http://192.168.4.1/#/status?verbose)
Hostname: ebus-e48a70
Up time: 108
Free heap: 122516
```
Die Auswahl als WiFi station/client ist zweifelsfrei gesetzt, wird aber auch nach einem Reboot nicht ausgeführt. Statt dessen erscheint die Meldung: 

`Status: begin failed, error: ESP_ERR_HTTP_CONNECT 
`

Das Log zeigt folgende Info:

```
ESP-ROM:esp32c3-api1-20210207
Build:Feb  7 2021
rst:0x15 (USB_UART_CHIP_RESET),boot:0xd (SPI_FAST_FLASH_BOOT)
Saved PC:0x403880e6
SPIWP:0xee
mode:DIO, clock div:1
load:0x3fcd5820,len:0x1188
load:0x403cc710,len:0x8bc
load:0x403ce710,len:0x2b98
entry 0x403cc710
```

Demnach komme ich einfach nicht weiter. Die letzte Version aus 2023 funktioniert (bei mir) nicht – und die neue aus 2024 läßt sich nicht installieren. 

Was mache ich falsch?

mfg Gerd

**Comment by wepanx** [comment]
Bei mir ähnlich. Mein Adapter war leider länger liegen geblieben. Nun bei der Ersteinrichtung keine Verbindung per Wifi abseits vom AP-Modus möglich. Startete mit Firmware 2023-05-01, nach Update auf 2024-01-06 selbes Bild. Beim Versuch die Nutzerdaten zu löschen (erase user data) oder die Firmware downzugraden, kommt dieselbe Fehlermeldung wie hier im Thread: 
> Failed to initialize. Try resetting your device or holding the BOOT button while clicking INSTALL.  


wifi scan/query ergibt kein Ergebnis. Ebenso wenig kann ich APs über das Webinterface vom Flasher/Terminal sehen. Eine Verbindung zu meinem AP ist nicht möglich.
AP-Modus funktioniert.

**Comment by cebulonde** [comment]
Ich habe das Problem gelöst.

Ursache waren offensichtlich mehrere Sonderzeichen im Zugangscode für meinen Router (Fritzbox 7490).  Nach einer Änderung auf "normale" Zeichen habe ich auf die Version 20240317 updaten können. Damit funktioniert der Zugriff auf den Adapter einwandfre!
